### PR TITLE
Improve outcome analysis for driver-only builds

### PR DESCRIPTION
**Pull request by mpg**
## Description

Improve outcome analysis by using an exclude list rather than an include list for test suites to compare, and making the reference configuration closer to the driver-based configuration.

## Status
**READY**

## Requires Backporting

NO  - the script is in development only, to test development-only features

## Requires ChangeLog

NO - this is a developer-only script

### CODE REVIEW

**Review by mprse — COMMENTED** [review]
Tested script locally and found one general problem: `test_psa_crypto_config_accel_hash_use_psa`
is now running `ssl-opt.sh` and `compat.sh` which are redundant for our case and take lots of time. We can comment out this part from the test component, but it is easy to forget. 

Additionally we will need more reference configurations, so maybe we could have keep reference implementations for hash, cipher, aead, etc. in the separate file.

Change looks good and works fine.

**Comment by mpg** [comment]
@mprse Thanks for your review!

> Tested script locally and found one general problem: `test_psa_crypto_config_accel_hash_use_psa` is now running `ssl-opt.sh` and `compat.sh` which are redundant for our case and take lots of time. We can comment out this part from the test component, but it is easy to forget.

Indeed, I added something for that.
 
> Additionally we will need more reference configurations, so maybe we could have keep reference implementations for hash, cipher, aead, etc. in the separate file.

Good point. I've given it some thought, and actually, I think this script should be only for the work in progress (so, hash so far, and very soon moving to cipher+aead). Work that's complete (so, hash as soon as #6437 is done) should actually be tested in the CI. I'll create issues about that.

**Comment by mpg** [comment]
Ok, I've created https://github.com/Mbed-TLS/mbedtls/issues/6445 to add that analysis for hashes to the CI immediately, and left a note in https://github.com/Mbed-TLS/mbedtls/issues/6337 to add it for cipher+aead when we're done.

**Comment by mprse** [comment]
The sign-off in commit is missing and CI is failing.

**Review by mprse — DISMISSED** [review]
LGTM.
Restarted Ci as it has glitched. 
(Test results provided in https://github.com/Mbed-TLS/mbedtls/pull/6447#issue-1413193492)

File: docs/architecture/psa-migration/outcome-analysis.sh [hunk 1]
```diff
@@ -31,13 +31,25 @@ DRIVER_COMPONENT=test_psa_crypto_config_accel_hash_use_psa
...
+    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_DETERMINISTIC_ECDSA
 }
-# Space-separated list of test suites of interest.
-SUITES="rsa pkcs1_v15 pk pkparse pkwrite"
+# Space-separated list of test suites to ignore:
+# if SSS is in that list, test_suite_SSS and test_suite_SSS.* are ignored.
+IGNORE="md mdx shax" # accelerated
+IGNORE="$IGNORE entropy hmac_drbg random" # disabled (ext. RNG)
+IGNORE="$IGNORE psa_crypto_init" # needs internal RNG
+IGNORE="$IGNORE hkdf" # disabled
```
Comment (AndrzejKurek) [inline docs/architecture/psa-migration/outcome-analysis.sh:50]: A bit of an explanation here would help.

File: tests/scripts/all.sh [hunk 2]
```diff
@@ -2063,11 +2063,16 @@ component_test_psa_crypto_config_accel_hash_use_psa () {
...
+    if [ "${SKIP_SSL_OPT_COMPAT_SH-unset}" = "unset" ]; then
+        msg "test: ssl-opt.sh, MBEDTLS_PSA_CRYPTO_CONFIG with accelerated hash and USE_PSA"
+        tests/ssl-opt.sh
 
-    msg "test: compat.sh, MBEDTLS_PSA_CRYPTO_CONFIG with accelerated hash and USE_PSA"
-    tests/compat.sh
+        msg "test: compat.sh, MBEDTLS_PSA_CRYPTO_CONFIG with accelerated hash and USE_PSA"
+        tests/compat.sh
+    else
+        echo "skip sh scripts"
```
Comment (AndrzejKurek) [inline tests/scripts/all.sh:2074]: More detail here would be great: "compat.sh and ssl-opt.sh skipped"

File: docs/architecture/psa-migration/outcome-analysis.sh [hunk 3]
```diff
@@ -81,10 +95,27 @@ record "reference"
...
+    for data in $data_files; do
+        suite=${data#test_suite_}
+        suite=${suite%.data}
+        suite_base=${suite%%.*}
+        case " $IGNORE " in
+            *" $suite_base "*) :;;
+            *) SUITES="$SUITES $suite";;
+        esac
+    done
+    make neat
```
Comment (AndrzejKurek) [inline docs/architecture/psa-migration/outcome-analysis.sh:116]: Wow, I didn't know that target :)

**Review by AndrzejKurek — CHANGES_REQUESTED** [review]
Just two cosmetic remarks.

**Review by AndrzejKurek — APPROVED** [review]

**Review by mprse — APPROVED** [review]

**Pull request merged**